src/DataTable/index.tsx

```diff
@@ -261,6 +262,7 @@ export const DataTable = <T extends Record<string, unknown>>(
     nextPage,
     previousPage,
     setPageSize,
+    totalResults,
```

Comment:
@Blakidarkness we're going to want to accept an optional incoming prop for `totalResults` that is a number. If that incoming prop is set we'll use that prop on L228 to setTotalResults instead of `data.length`. If it isn't set we can use `data.length` as the default.

The reason for this is that someone using an asynchronous pagination system would have more total results than the data currently in the table. If there are 2000 items in the API we want to set that as the total number instead of the 10 they're loading initially.